ui/src/androidTest/java/com/hyperwallet/android/transfermethod/ui/ListReceiptsTest.java

```diff
@@ -0,0 +1,150 @@
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(2, hasDescendant(withText("USD")))));
+
+        onView(withId(R.id.list_receipts))
+                .check(matches(atPosition(3, hasDescendant(withText("December 2017")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(3,
+                hasDescendant(withText(com.hyperwallet.android.receipt.R.string.debit)))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(3, hasDescendant(withText("Card Load")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(3, hasDescendant(withText("- 18.05")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(3, hasDescendant(withText("December 01, 2017")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(3, hasDescendant(withText("USD")))));
+
+        onView(withId(R.id.list_receipts)).check(new RecyclerViewCountAssertion(4));
+    }
+
+    @Test
+    public void testListReceipt_userHasSingleReceipt() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("receipt_list_single_receipt_response.json")).mock();
+
+        // run test
+        mActivityTestRule.launchActivity(null);
+
+        // assert
+        onView(allOf(instanceOf(TextView.class), withParent(withId(R.id.toolbar))))
+                .check(matches(withText(R.string.title_activity_receipt_list)));
+        onView(withId(R.id.list_receipts)).check(matches(isDisplayed()));
+
+        onView(withId(R.id.list_receipts))
+                .check(matches(atPosition(0, hasDescendant(withText("November 2017")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(0,
+                hasDescendant(withText(com.hyperwallet.android.receipt.R.string.credit)))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(0, hasDescendant(withText("Payment")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(0, hasDescendant(withText("+ 25.00")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(0, hasDescendant(withText("November 02, 2017")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(0, hasDescendant(withText("CAD")))));
+
+        onView(withId(R.id.list_receipts)).check(new RecyclerViewCountAssertion(1));
+    }
+
+    @Test
+    public void testListReceipt_userHasNoReceipts() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_NO_CONTENT).withBody("").mock();
+
+        // run test
+        mActivityTestRule.launchActivity(null);
+
+        onView(withId(R.id.toolbar)).check(matches(isDisplayed()));
+        onView(allOf(instanceOf(TextView.class), withParent(withId(R.id.toolbar))))
+                .check(matches(withText(R.string.title_activity_receipt_list)));
+        //todo: check empty view when it will be ready
+    }
```

Comment:
Please add another test case for lazy loading - testListReceipts_verifyTransactionsLoadedUponScrolling()